x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/mapper/EvaluatorMapper.java

```diff
@@ -27,6 +28,22 @@
  */
 public interface EvaluatorMapper {
     /**
+     * <p>
+     * Note for implementors:
+     * If you are implementing this function, you should call the passed-in
+     * lambda on your children, after doing any other manipulation (casting,
+     * etc.) necessary.
+     * </p>
+     * <p>
+     * Note for Callers:
+     * If you are attempting to call this method, and you have an
+     * {@link Expression} and a {@link org.elasticsearch.xpack.esql.planner.Layout},
+     * you likely want to call {@link org.elasticsearch.xpack.esql.evaluator.EvalMapper#toEvaluator(Expression, Layout)}
+     * instead.  On the other hand, if you already have something that
+     * looks like the parameter for this method, you should call this method
+     * with that function.
+     * </p>
+     * <p>
```

Comment:
@nik9000 I tried to make this as clear as possible.  Please tell me if this reads well to you.

Reply:
This is perfect.